modules/drivers/snowflake/src/metabase/driver/snowflake.clj

```diff
@@ -575,3 +578,16 @@
 (defmethod sql-jdbc.execute/set-parameter [:snowflake java.time.ZonedDateTime]
   [driver ps i t]
   (sql-jdbc.execute/set-parameter driver ps i (t/sql-timestamp (t/with-zone-same-instant t (t/zone-id "UTC")))))
+
+
+;;; ------------------------------------------------- User Impersonation --------------------------------------------------
+
+(defmethod qp.util/set-role-statement :snowflake
+  [_ role]
+  (format "USE ROLE %s;" role))
```

Comment:
Unfortunately prepared statements can't be used for `USE ROLE` commands so we have to construct the SQL string manually. But I think we're safe from SQL injections because 1. impersonation configs & user attributes can only be set by admins (or come via SSO) and 2. snowflake only allows one statement to be executed per command